ste/securityInfoPersistenceManager.go

```diff
@@ -2,127 +2,87 @@ package ste
 }
 
-func (sipm *securityInfoPersistenceManager) PutServiceURL(serviceURL azfile.ServiceURL) {
-	fURLParts := common.NewGenericResourceURLParts(serviceURL.URL(), common.ELocation.File())
-
-	sipm.sipmMu.Lock()
-	defer sipm.sipmMu.Unlock()
-
-	sipm.serviceURLs[fURLParts.GetAccountName()] = serviceURL
-}
-
 // Technically, yes, GetSDDLFromID can be used in conjunction with PutSDDL.
 // Being realistic though, GetSDDLFromID will only be called when downloading,
 // and PutSDDL will only be called when uploading/doing S2S.
-func (sipm *securityInfoPersistenceManager) PutSDDL(acctName, shareName, sddlString string, p pipeline.Pipeline) (string, error) {
-	sipm.sipmMu.Lock()
-	defer sipm.sipmMu.Unlock()
+func (sipm *securityInfoPersistenceManager) PutSDDL(sddlString string, shareURL azfile.ShareURL) (string, error) {
+	fileURLParts := azfile.NewFileURLParts(shareURL.URL())
+	fileURLParts.DirectoryOrFilePath = ""
+	rawfURL := fileURLParts.URL()
+
+	sddlKey := rawfURL.String() + "|SDDL|" + sddlString
 
 	// First, let's check the cache for a hit or miss.
 	// These IDs are per share, so we use a share-unique key.
 	// The SDDL string will be consistent from a local source.
-	id, ok := sipm.cache.Get(acctName + "|SHARE|" + shareName + "|SDDL|" + sddlString)
-
-	fmt.Println(acctName + "|SHARE|" + shareName + "|SDDL|" + sddlString)
+	id, ok := sipm.cache.Get(sddlKey)
 
 	if ok {
-		fmt.Println("cache HIT!")
 		return id.(string), nil
 	}
 
-	parsedSDDL, err := sddl.ParseSDDL(sddlString)
-
-	if err != nil {
-		return "", err
-	}
-
-	// No extra work is needed to make SIDs portable, because SID.String() calls SID.ToPortable() for us.
-	upSDDL := parsedSDDL.PortableString()
-
-	serviceURL, ok := sipm.serviceURLs[acctName]
-
-	if !ok {
-		return "", fmt.Errorf("service URL wasn't found for account %s", acctName)
-	}
-
-	serviceURL = serviceURL.WithPipeline(p)
-
-	shareURL := serviceURL.NewShareURL(shareName)
-
-	cResp, err := shareURL.CreatePermission(sipm.ctx, upSDDL)
+	cResp, err := shareURL.CreatePermission(sipm.ctx, sddlString)
 
 	if err != nil {
 		return "", err
 	}
 
-	key := cResp.FilePermissionKey()
+	permKey := cResp.FilePermissionKey()
 
-	sipm.cache.Add(acctName+"|SHARE|"+shareName+"|SDDL|"+sddlString, key)
+	sipm.cache.Add(sddlKey, permKey)
 
-	return key, nil
+	return permKey, nil
 }
 
-func (sipm *securityInfoPersistenceManager) GetSDDLFromID(acctName, shareName, id string, p pipeline.Pipeline) (string, error) {
-	sipm.sipmMu.Lock()
-	defer sipm.sipmMu.Unlock()
+func (sipm *securityInfoPersistenceManager) GetSDDLFromID(id string, shareURL azfile.ShareURL) (string, error) {
+	fileURLParts := azfile.NewFileURLParts(shareURL.URL())
+	fileURLParts.DirectoryOrFilePath = ""
+	rawfURL := fileURLParts.URL()
+
+	sddlKey := rawfURL.String() + "|ID|" + id
 
 	// fetch from the cache
 	// The SDDL string will be consistent from a local source.
-	perm, ok := sipm.cache.Get(acctName + "|SHARE|" + shareName + "|ID|" + id)
+	perm, ok := sipm.cache.Get(sddlKey)
```

Comment:
I think you still need the locks, around all access to simp.cache, in both this routine and the otherone.  The change in account URL handling did remove _one_ thing that needed to be locked... but it didn't remove the other. Right?

Reply:
True, the lru cache isn't safe for concurrent access.